src/anomalib/trainer/utils/thresholder.py

```diff
@@ -85,17 +69,45 @@ def update(self, outputs: STEP_OUTPUT) -> None:
         Args:
             outputs (STEP_OUTPUT): Step outputs.
         """
-        if self.threshold_method == ThresholdMethod.ADAPTIVE:
-            self._update_thresholds(self.trainer.image_threshold, self.trainer.pixel_threshold, outputs)
-
-    @staticmethod
-    def _update_thresholds(
-        image_metric: AnomalyScoreThreshold,
-        pixel_metric: AnomalyScoreThreshold,
-        outputs: STEP_OUTPUT,
-    ) -> None:
-        image_metric.cpu()
-        image_metric.update(outputs["pred_scores"], outputs["label"].int())
-        if "mask" in outputs.keys() and "anomaly_maps" in outputs.keys():
-            pixel_metric.cpu()
-            pixel_metric.update(outputs["anomaly_maps"], outputs["mask"].int())
+        image_threshold = self.trainer.image_threshold
+        pixel_threshold = self.trainer.pixel_threshold
+        image_threshold.cpu()
+        image_threshold.update(outputs["pred_scores"], outputs["label"].int())
+        if (
+            self.trainer.task_type != TaskType.CLASSIFICATION
+            and "anomaly_maps" in outputs.keys()
+            and "mask" in outputs.keys()
+        ):
+            pixel_threshold.cpu()
+            # TODO this should use bounding boxes for detection task type
+            pixel_threshold.update(outputs["anomaly_maps"], outputs["mask"].int())
+
+    def _get_threshold_metric(self, threshold_method: dict | None) -> BaseAnomalyThreshold:
+        """Get instantiated threshold metric.
+
+        Args:
+            threshold_method (dict | None): Threshold method. Defaults to F1AdaptiveThreshold.
+
+        Returns:
+            Instantiated threshold metric.
+        """
+        threshold_metric: BaseAnomalyThreshold
+        if threshold_method is None:
+            threshold_metric = F1AdaptiveThreshold()
+        else:
+            _class_path = threshold_method.get("class_path", "F1AdaptiveThreshold")
+            try:
+                if len(_class_path.split(".")) > 1:  # When the entire class path is provided
+                    threshold_module = import_module(".".join(_class_path.split(".")[:-1]))
+                    _threshold_class = getattr(threshold_module, _class_path.split(".")[-1])
+                else:
+                    threshold_module = import_module("anomalib.utils.metrics.thresholding")
+                    _threshold_class = getattr(threshold_module, _class_path)
+            except (AttributeError, ModuleNotFoundError) as exception:
+                raise ValueError(f"Threshold class {_class_path} not found") from exception
+
+            init_args = threshold_method.get("init_args")
+            init_args = init_args if init_args is not None else {}
+            threshold_metric = _threshold_class(**init_args)
+
+        return threshold_metric
```

Comment:
I'm guessing this will change with the new CLI? Then I'll ignore this part

Reply:
Yeah. I'll add TODO to this class